checkout_and_build_pr.sh

```diff
@@ -0,0 +1,242 @@
+    BASE_DIR=$2
+
+    echo "Passed branch dir to make ${BRANCH_DIR_TO_MAKE}"
+
+    BASE_DIR="${BASE_DIR}/${BRANCH_DIR_TO_MAKE}"
+
+    # TODO:  What to do if dir alerady exists? rm it, fail etc.. 
+
+    echo "Making new base dir ${BASE_DIR}"
+    mkdir $BASE_DIR
+
+    cd $BASE_DIR
+    echo "Now in new DIR: ${PWD}"
+}
+
+function wait_until_containers_are_running_or_timeout() {
+    WAIT_FOR_DOCKER_UP_TIMEOUT_SECONDS=60
+    starting_containers=""
+
+    while true
+    do
+        starting_containers = $("docker ps | grep -E 'starting|unhealthy'")
+        echo -e "${starting_containers}"
+
+        if ["$starting_containers" = ""] ; then
+            echo "Looks like containers are up"
+            return
+        fi
+            
+        # Wait and try again in a second
+        sleep 1
+
+        echo "RES: ${starting_containers}"
+    
+        i=$[$i+1]
+    done
+
+    echo -e "\033[1;31m  ERROR: Docker Images did not come up in expected time \n ${starting_containers} \033[0m"
+    exit 1
+}
+
+
+########################################################################################################################
+#
+#         START OF SCRIPT
+#
+########################################################################################################################
+
+# Internal Variable, will use to record time
+SECONDS=0
+
+docker ps | grep -E 'starting|unhealthy'
+
+
+exit 1
+
+# Check Branch name is set
+if [ -z "$BRANCH_NAME" ]; then
+    echo "You Must set BRANCH_NAME=<branch to test>, use main, if you want main"
+    exit 2;
+fi
+
+# Base Dir to Checkout To
+if [ -z "$BASE_DIR" ]; then
+    echo "You Must set BASE_DIR=<your base dir>.  BASE_DIR does not need the branch name, that will be created"
+    exit 2;
+fi
+
+# Create the baseDir
+createNewBaseDir $BRANCH_NAME $BASE_DIR
+
+if [ "$SKIP_TESTS" = true ] ; then
+    echo "Script will Skip Tests"
+else
+    echo "Script will be Running Tests"
+fi
+
+# Kill and remove running containers, Flag to disable exists
+killOffRunningDocker
+
+checkout_repo_branch "ssdc-rm-docker-dev" $BRANCH_NAME
+execute_and_record_command "pipenv install --dev" true
+execute_and_record_command "make up" true
+popd
+
+wait_until_containers_are_running_or_timeout
+
+exit 0
+
+
+########################################################################################################################
+#  Build, Install and Test DDL and Applications
+########################################################################################################################
+
+# TODO: Maven -Dmaven.test.skip=true stops maven running tests.  However the pre & post integration steps are still 
+# run. Ideally we'd skip these too.  May require pom changes though.
+
+MVN_INSTALL_TEST_CMD="mvn clean install"
+MVN_INSTALL_ONLY_CMD="mvn clean install -Dmaven.test.skip=true -DdockerCompose.skip=true"
+
+# Install DDL
+checkout_and_build_repo_branch "ssdc-rm-ddl" $BRANCH_NAME "make dev-build" "make dev-build"
```

Comment:
This DDL build failed for me because I didn't have a build of ssdc-shared-sample-validation.  Can you add that in before this?